llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -421,27 +466,132 @@ Expected<SectionRef> SectionRef::create(MCCASBuilder &MB,
   return get(B->build());
 }
 
+Expected<SmallVector<char, 0>> MCCASReader::applyRelocationAddends(
+    const SmallVector<cas::ObjectRef> &Refs, StringRef RelocData,
+    StringLiteral KindString, StringRef SectionName,
+    bool &IsPaddingRefPresent) {
+  if (auto E = decodeRelocations(*this, RelocData))
+    return std::move(E);
+  SmallVector<char, 0> SectionData;
+  auto RelocRef = MCObjectProxy::get(Schema, Schema.CAS.getProxy(Refs[0]));
+  if (!RelocRef)
+    return RelocRef.takeError();
+  if (auto R = RelocAddendRef::Cast(*RelocRef)) {
```

Comment:
Let's use an early return here, this block is huge, making reviewing a bit challenging

```cpp
auto R = RelocAddendRef::Cast(*RelocRef));
if (!R) return error ...
```

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code